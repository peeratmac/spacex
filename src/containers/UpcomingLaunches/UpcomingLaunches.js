import React, { Component } from 'react';
import { getFutureElonMuskDreams } from '../../apiCalls';
import LaunchCard from '../../containers/LaunchCard/LaunchCard';
import { connect } from 'react-redux';
import './UpcomingLaunches.css';
import { addUpcomingSpaceData } from '../../actions';

export const UpcomingLaunches = ({ upcomingLaunches, saveFavorite }) => {
  const launchesData = upcomingLaunches.map((launch, index) => {
    return (
      <LaunchCard launch={launch} saveFavorite={saveFavorite} key={index} />
    );
  });

  return (
    <div>
      <div className='grid-3'>{launchesData}</div>
    </div>
  );
};

export const mapStateToProps = state => ({
  upcomingLaunches: state.upcomingSpaceData
});

export default connect(mapStateToProps, null)(UpcomingLaunches);
