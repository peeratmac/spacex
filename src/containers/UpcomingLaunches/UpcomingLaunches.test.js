import React from 'react';
import { shallow } from 'enzyme';
import { UpcomingLaunches } from './UpcomingLaunches';

describe('UPCOMING LAUNCHES', () => {
  const mockUpcomingLaunchesData = [
    {
      flight_number: 88,
      mission_name: 'Crew Dragon In Flight Abort Test',
      mission_id: ['EE86F74'],
      launch_year: '2020',
      launch_date_unix: 1579352400,
      launch_date_utc: '2020-01-18T13:00:00.000Z',
      launch_date_local: '2020-01-18T08:00:00-05:00',
      is_tentative: false,
      tentative_max_precision: 'hour',
      tbd: false,
      launch_window: null,
      rocket: {
        rocket_id: 'falcon9',
        rocket_name: 'Falcon 9',
        rocket_type: 'FT',
        first_stage: {
          cores: [
            {
              core_serial: 'B1046',
              flight: 4,
              block: 5,
              gridfins: null,
              legs: null,
              reused: true,
              land_success: null,
              landing_intent: null,
              landing_type: null,
              landing_vehicle: null
            }
          ]
        },
        second_stage: {
          block: null,
          payloads: [
            {
              payload_id: 'Crew Dragon In Flight Abort Test',
              norad_id: [],
              cap_serial: null,
              reused: true,
              customers: ['NASA (CCtCap)'],
              nationality: 'United States',
              manufacturer: 'SpaceX',
              payload_type: 'Crew Dragon',
              payload_mass_kg: null,
              payload_mass_lbs: null,
              orbit: 'SO',
              orbit_params: {
                reference_system: 'geocentric',
                regime: 'sub-orbital',
                longitude: null,
                semi_major_axis_km: null,
                eccentricity: null,
                periapsis_km: null,
                apoapsis_km: null,
                inclination_deg: null,
                period_min: null,
                lifespan_years: null,
                epoch: null,
                mean_motion: null,
                raan: null,
                arg_of_pericenter: null,
                mean_anomaly: null
              },
              mass_returned_kg: null,
              mass_returned_lbs: null,
              flight_time_sec: null,
              cargo_manifest: null,
              uid: 'UerI6qmZTU2Fx2efDFm3QQ=='
            }
          ]
        },
        fairings: null
      },
      ships: [],
      telemetry: {
        flight_club: null
      },
      launch_site: {
        site_id: 'ksc_lc_39a',
        site_name: 'KSC LC 39A',
        site_name_long: 'Kennedy Space Center Historic Launch Complex 39A'
      },
      launch_success: null,
      links: {
        mission_patch: null,
        mission_patch_small: null,
        reddit_campaign:
          'https://www.reddit.com/r/spacex/comments/ek7eny/in_flight_abort_test_launch_campaign_thread',
        reddit_launch: null,
        reddit_recovery: null,
        reddit_media: null,
        presskit: null,
        article_link: null,
        wikipedia: null,
        video_link: null,
        youtube_id: null,
        flickr_images: []
      },
      details:
        "SpaceX will launch a Crew Dragon capsule from LC-39A, KSC on a fully fueled Falcon 9 rocket and then trigger the launch escape system during the period of maximum dynamic pressure. As part of NASA'a Commercial Crew Integrated Capability program (CCiCap) this test will contribute valuable data to help validate Crew Dragon and its launch abort system. The Crew Dragon will be recovered by GO Searcher after splashdown in the Atlantic Ocean. This flight does not go to orbit. The booster and upper stage are expected to break up following capsule separation and there will be no landing attempt.",
      upcoming: true,
      static_fire_date_utc: null,
      static_fire_date_unix: null,
      timeline: null,
      crew: null,
      last_date_update: '2019-11-27T01:51:12.000Z',
      last_ll_launch_date: null,
      last_ll_update: null,
      last_wiki_launch_date: '2019-12-01T00:00:00.000Z',
      last_wiki_revision: '638b6fa7-10b8-11ea-bc31-0e65898001b1',
      last_wiki_update: '2019-11-27T01:51:12.000Z',
      launch_date_source: 'wiki'
    }
  ];

  it('should match the UpcomingLaunches Snapshot UI', () => {
    let wrapper = shallow(
      <UpcomingLaunches
        upcomingLaunches={mockUpcomingLaunchesData}
        saveFavorite={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
