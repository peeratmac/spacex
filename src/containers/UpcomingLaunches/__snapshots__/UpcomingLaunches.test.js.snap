// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UPCOMING LAUNCHES should match the UpcomingLaunches Snapshot UI 1`] = `
<div>
  <div
    className="grid-3"
  >
    <LaunchCard
      key="0"
      launch={
        Object {
          "crew": null,
          "details": "SpaceX will launch a Crew Dragon capsule from LC-39A, KSC on a fully fueled Falcon 9 rocket and then trigger the launch escape system during the period of maximum dynamic pressure. As part of NASA'a Commercial Crew Integrated Capability program (CCiCap) this test will contribute valuable data to help validate Crew Dragon and its launch abort system. The Crew Dragon will be recovered by GO Searcher after splashdown in the Atlantic Ocean. This flight does not go to orbit. The booster and upper stage are expected to break up following capsule separation and there will be no landing attempt.",
          "flight_number": 88,
          "is_tentative": false,
          "last_date_update": "2019-11-27T01:51:12.000Z",
          "last_ll_launch_date": null,
          "last_ll_update": null,
          "last_wiki_launch_date": "2019-12-01T00:00:00.000Z",
          "last_wiki_revision": "638b6fa7-10b8-11ea-bc31-0e65898001b1",
          "last_wiki_update": "2019-11-27T01:51:12.000Z",
          "launch_date_local": "2020-01-18T08:00:00-05:00",
          "launch_date_source": "wiki",
          "launch_date_unix": 1579352400,
          "launch_date_utc": "2020-01-18T13:00:00.000Z",
          "launch_site": Object {
            "site_id": "ksc_lc_39a",
            "site_name": "KSC LC 39A",
            "site_name_long": "Kennedy Space Center Historic Launch Complex 39A",
          },
          "launch_success": null,
          "launch_window": null,
          "launch_year": "2020",
          "links": Object {
            "article_link": null,
            "flickr_images": Array [],
            "mission_patch": null,
            "mission_patch_small": null,
            "presskit": null,
            "reddit_campaign": "https://www.reddit.com/r/spacex/comments/ek7eny/in_flight_abort_test_launch_campaign_thread",
            "reddit_launch": null,
            "reddit_media": null,
            "reddit_recovery": null,
            "video_link": null,
            "wikipedia": null,
            "youtube_id": null,
          },
          "mission_id": Array [
            "EE86F74",
          ],
          "mission_name": "Crew Dragon In Flight Abort Test",
          "rocket": Object {
            "fairings": null,
            "first_stage": Object {
              "cores": Array [
                Object {
                  "block": 5,
                  "core_serial": "B1046",
                  "flight": 4,
                  "gridfins": null,
                  "land_success": null,
                  "landing_intent": null,
                  "landing_type": null,
                  "landing_vehicle": null,
                  "legs": null,
                  "reused": true,
                },
              ],
            },
            "rocket_id": "falcon9",
            "rocket_name": "Falcon 9",
            "rocket_type": "FT",
            "second_stage": Object {
              "block": null,
              "payloads": Array [
                Object {
                  "cap_serial": null,
                  "cargo_manifest": null,
                  "customers": Array [
                    "NASA (CCtCap)",
                  ],
                  "flight_time_sec": null,
                  "manufacturer": "SpaceX",
                  "mass_returned_kg": null,
                  "mass_returned_lbs": null,
                  "nationality": "United States",
                  "norad_id": Array [],
                  "orbit": "SO",
                  "orbit_params": Object {
                    "apoapsis_km": null,
                    "arg_of_pericenter": null,
                    "eccentricity": null,
                    "epoch": null,
                    "inclination_deg": null,
                    "lifespan_years": null,
                    "longitude": null,
                    "mean_anomaly": null,
                    "mean_motion": null,
                    "periapsis_km": null,
                    "period_min": null,
                    "raan": null,
                    "reference_system": "geocentric",
                    "regime": "sub-orbital",
                    "semi_major_axis_km": null,
                  },
                  "payload_id": "Crew Dragon In Flight Abort Test",
                  "payload_mass_kg": null,
                  "payload_mass_lbs": null,
                  "payload_type": "Crew Dragon",
                  "reused": true,
                  "uid": "UerI6qmZTU2Fx2efDFm3QQ==",
                },
              ],
            },
          },
          "ships": Array [],
          "static_fire_date_unix": null,
          "static_fire_date_utc": null,
          "tbd": false,
          "telemetry": Object {
            "flight_club": null,
          },
          "tentative_max_precision": "hour",
          "timeline": null,
          "upcoming": true,
        }
      }
      saveFavorite={[MockFunction]}
    />
  </div>
</div>
`;
